cpu/cortexm_common/vectors_cortexm.c

```diff
@@ -134,7 +134,8 @@ void hard_fault_default(void)
         "mrs r0, psp                        \n" /*   r0 = psp                 */
         " out:                              \n" /* }                          */
         "mov r1, #0                         \n" /* corrupted = false          */
-        "cmp r0, sp                         \n" /* If msp is active stack-    */
+        "mov r2, sp                         \n" /* Cmp with SP is deprecated  */
+        "cmp r0, r2                         \n" /* If msp is active stack-    */
```

Comment:
Couldn't we use `cmp sp, r0`? `sp` is only deprecated as second operant for `cmp`. Compare https://github.com/RIOT-OS/RIOT/issues/3664#issuecomment-132743467